cloudflare/resource_cloudflare_waf_package.go

```diff
@@ -0,0 +1,255 @@
+	if packageName == "" && packageID == "" {
+		return fmt.Errorf("either package name or ID must be provided")
+	}
+
+	if packageID == "" {
+		pkg, err := packageFromName(client, zoneID, packageName)
+		if err != nil {
+			return err
+		}
+		packageID = pkg.ID
+	}
+
+	pkg, err := client.WAFPackage(zoneID, packageID)
+	if err != nil {
+		return fmt.Errorf("Unable to find WAF Package %s", packageID)
+	}
+
+	d.Set("zone_id", zoneID)
+	d.Set("package_id", packageID)
+	d.Set("package_name", pkg.Name)
+	d.Set("sensitivity", sensitivity)
+	d.Set("action_mode", actionMode)
+
+	// Set the ID to the package_id parameter passed in from the user.
+	// All WAF packages already exist so we already know the package_id.
+	//
+	// This is a work around as we are not really "creating" a WAF Package,
+	// only associating it with our terraform config for future updates.
+	d.SetId(packageID)
+
+	if pkg.Sensitivity != sensitivity || pkg.ActionMode != actionMode {
+		err = resourceCloudflareWAFPackageUpdate(d, meta)
+		if err != nil {
+			d.SetId("")
+			return err
+		}
+	}
+
+	return nil
+}
+
+func resourceCloudflareWAFPackageDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+
+	packageID := d.Get("package_id").(string)
+	zoneID := d.Get("zone_id").(string)
+
+	pkg, err := client.WAFPackage(zoneID, packageID)
+	if err != nil {
+		return err
+	}
+
+	// Can't delete WAF Package so instead reset it to default
+	schema := resourceCloudflareWAFPackage().Schema
+	defaultSensitivity := schema["sensitivity"].Default.(string)
+	defaultActionMode := schema["action_mode"].Default.(string)
+
+	if pkg.Sensitivity != defaultSensitivity || pkg.ActionMode != defaultActionMode {
+		options := cloudflare.WAFPackageOptions{
+			Sensitivity: defaultSensitivity,
+			ActionMode:  defaultActionMode,
+		}
+
+		_, err = client.UpdateWAFPackage(zoneID, packageID, options)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func resourceCloudflareWAFPackageUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+
+	packageID := d.Get("package_id").(string)
+	zoneID := d.Get("zone_id").(string)
+	sensitivity := d.Get("sensitivity").(string)
+	actionMode := d.Get("action_mode").(string)
+
+	options := cloudflare.WAFPackageOptions{
+		Sensitivity: sensitivity,
+		ActionMode:  actionMode,
+	}
+
+	_, err := client.UpdateWAFPackage(zoneID, packageID, options)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func resourceCloudflareWAFPackageImport(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	client := meta.(*cloudflare.API)
+
+	// split the id so we can lookup
+	idAttr := strings.SplitN(d.Id(), "/", 2)
+	var zoneName string
+	var packageNameOrID string
+	if len(idAttr) == 2 {
+		zoneName = idAttr[0]
+		packageNameOrID = idAttr[1]
+	} else {
+		return nil, fmt.Errorf("invalid id (\"%s\") specified, should be in format \"zoneName/PackageNameOrID\" for import", d.Id())
```

Comment:
As we're only supporting zone ID we can get rid of the zone name references.

```suggestion
		return nil, fmt.Errorf("invalid id (\"%s\") specified, should be in format \"zoneID/PackageNameOrID\" for import", d.Id())
```